Eggstravaganza/Assets/TestNetcodeExample/_Scripts/Player/PlayerTransform.cs

```diff
@@ -0,0 +1,94 @@
+using Unity.Netcode;
+using UnityEngine;
+
+/// <summary>
+/// An example network serializer with both server and owner authority.
+/// Love Tarodev
+/// </summary>
+public class PlayerTransform : NetworkBehaviour {
+    /// <summary>
+    /// A toggle to test the difference between owner and server auth.
+    /// </summary>
+    [SerializeField] private bool _serverAuth;
+    [SerializeField] private float _cheapInterpolationTime = 0.1f;
+
+    private NetworkVariable<PlayerNetworkState> _playerState;
+    private Rigidbody _rb;
+
+    private void Awake() {
+        _rb = GetComponent<Rigidbody>();
+
+        var permission = _serverAuth ? NetworkVariableWritePermission.Server : NetworkVariableWritePermission.Owner;
+        _playerState = new NetworkVariable<PlayerNetworkState>(writePerm: permission);
+    }
+
+    public override void OnNetworkSpawn() {
+        if (!IsOwner) Destroy(transform.GetComponent<TestPlayerController>());
```

Comment:
just curious, but did you ever run into issues with this line? when i was trying this in a different scene/branch, Destroying gave me non-matching objects across clients (components mismatch etc) so i ended up setting enabled = false, but yours works!